src/dump-xml.c

```diff
@@ -35,68 +42,434 @@ typedef struct {
 	{"symbolrate", {SYS_DVBS, SYS_DVBS2, SYS_DSS, SYS_TURBO,
 			SYS_DVBC_ANNEX_A, SYS_DVBC_ANNEX_B, SYS_DVBC_ANNEX_C,
-			0},
-	 NO_AUTO},
-
-	// see S2_satellite_delivery_system_descriptor
+			0}, NO_AUTO},
 	{"multiple_input_stream_flag", {SYS_DVBS2, 0}, 0},
 	{"scrambling_sequence_selector", {SYS_DVBS2, 0}, 0},
 	{"scrambling_sequence_index", {SYS_DVBS2, 0}, 0},
-	{"input_stream_id", {SYS_DVBS2, 0}, NO_AUTO},
+	{"input_stream_id", {SYS_DVBT2, SYS_DVBS2, 0}, NO_AUTO},
 	{"pilot", {SYS_DVBS2, 0}, PILOT_AUTO},
-
-	//
-	{"interleave", {SYS_DTMB, 0}, INTERLEAVING_AUTO},
-
-	// C2_delivery_system_descriptor
 	{"data_slice_id", {SYS_DVBC2, 0}, 0},
-	{"C2_System_tuning_frequency_type", {SYS_DVBC2, 0},
+	{"C2_tuning_frequency_type", {SYS_DVBC2, 0},
 	 C2_SYSTEM_CENTER_FREQUENCY},
 	{"active_OFDM_symbol_duration", {SYS_DVBC2, 0}, FFT_4K_8MHZ},
 	{"", {0}, NO_AUTO},
 };
 
 #define NUM_XML_PARAMS (sizeof(xml_params) / sizeof(xml_params[0]))
 
+char *CaName(int id)
+{
+// function taken from w_scan_cpp (https://www.gen2vdr.de/wirbel/w_scan_cpp/index2.html)
+// int minor = id & 0xFF;
+	int major = id >> 8;
+
+	if (major == 0x01)
+		return "Seca Mediaguard";
+	else if (major == 0x05)
+		return "Viaccess";
+	else if (major == 0x06)
+		return "Irdeto";
+	else if (major == 0x07)
+		return "DigiCipher-2";
+	else if (major == 0x09)
+		return "NDS/Cisco Videoguard";
+	else if (major == 0x0B)
+		return "Conax";
+	else if (major == 0x0D)
+		return "Cryptoworks";
+	else if (major == 0x0E)
+		return "PowerVu";
+	else if (major == 0x10)
+		return "RAS (Remote Authorization System)";
+	else if (major == 0x17)
+		return "Betaresearch (or Tunnel)";
+	else if (major == 0x18)
+		return "Nagravision";
+	else if (major == 0x22)
+		return "Codicrypt";
+	else if (major == 0x26)
+		return "BISS";
+	else if (id == 0x0464)
+		return "EuroDec";
+	else if (id == 0x2719)
+		return "InCrypt Cas";
+	else if (id == 0x4347)
+		return "Crypton";
+	else if (id == 0x4800)
+		return "Accessgate";
+	else if (id == 0x4900)
+		return "China Crypt";
+	else if (id == 0x4A02)
+		return "Tongfang";
+	else if (id == 0x4A10)
+		return "EasyCas";
+	else if (id == 0x4A20)
+		return "AlphaCrypt";
+	else if (id == 0x4A60)
+		return "Skycrypt / Neotioncrypt / Neotion SHL";
+	else if (id == 0x4A61)
+		return "Skycrypt / Neotioncrypt / Neotion SHL";
+	else if (id == 0x4A63)
+		return "Skycrypt / Neotioncrypt / Neotion SHL";
+	else if (id == 0x4A70)
+		return "Dreamcrypt";
+	else if (id == 0x4A80)
+		return "ThalesCrypt";
+	else if (id == 0x4AA1)
+		return "KeyFly";
+	else if (id == 0x4ABF)
+		return "CTI-CAS";
+	else if (id == 0x4AD0)
+		return "X-Crypt";
+	else if (id == 0x4AD1)
+		return "X-Crypt";
+	else if (id == 0x4AD4)
+		return "OmniCrypt";
+	else if (id == 0x4AE0)
+		return "DRE-Crypt";
+	else if (id == 0x4AE1)
+		return "DRE-Crypt";
+	else if (id == 0x4AE4)
+		return "CoreCrypt";
+	else if (id == 0x4AEA)
+		return "Cryptoguard";
+	else if (id == 0x4AEB)
+		return "Abel Quintic";
```

Comment:
else if (id == 0x4AEE)
		return "Bulcrypt";

Reply:
Can you make a new PR?

Reply:
> Can you make a new PR?

Yes there is also missing 
CAID 1010
and Verimatrix variants
but there is something missing to be complete 
https://trac.streamboard.tv/oscam/wiki/CardsList
old resource not updated for long time

----
CAID ID is OK but Provider ID is missing
aka:
`4AEE:000000
`and
`4AEE:000001
`